### PR TITLE
[rllib] Rollout should preprocess observations; some cleanups 

### PR DESCRIPTION
**Pull request by ericl**
## What do these changes do?

From https://groups.google.com/forum/#!topic/ray-dev/u-gybKK6-Ns

### CODE REVIEW

**Comment by richardliaw** [comment]
Can you add a test for this?

**Comment by ericl** [comment]
Test added

File: python/ray/rllib/test/test_nested_spaces.py [hunk 1]
```diff
@@ -340,6 +341,21 @@ def testMultiAgentComplexSpaces(self):
...
+        path = agent.save()
+        agent.stop()
+
+        # Test train works on restore
+        agent2 = PGAgent(env="nested")
+        agent2.restore(path)
+        agent2.train()
+
+        # Test rollout works on restore
+        rollout(agent2, "nested", 100)
```
Comment (richardliaw) [inline python/ray/rllib/test/test_nested_spaces.py]: wait this doesn't test that the preprocessing is handled correctly right?

Comment (ericl) [reply python/ray/rllib/test/test_nested_spaces.py]: It crashes if it doesnt.

**Review by richardliaw — APPROVED** [review]

**Pull request merged**